[TAP 8] Update TAP 8 for clarity and changes in the spec

mnm678 opened the pull request:
Notably this pr:
* specifies that rotate files will be included in snapshot metadata
* fixes/clarifies references to TAP 3
* Removes references to rejected TAPs
* Simplifies the TAP 8 mechanism by not requiring role name changes

JustinCappos reviewed (dismissed):
Looks fine to me, but it certainly is a mismatch of edits.

mnm678 commented:
> Looks fine to me, but it certainly is a mismatch of edits.

It's a collection of minor edits from re-reading this after 4 years of changes to TUF. If there's anything controversial (which I tried to avoid) I can move that into a different pr.

joshuagl commented on tap8.md:
```diff
@@ -196,31 +186,30 @@ The second suffix, referred to as 'PREV' is a SHA256 hash of the previous
+Timestamp and snapshot keys can rotate as
+well, leading to ROLE.rotate.ID.PREV files, where ID and PREV are as described above. Each file is
+signed by a quorum of old keys, and contains the new keys. During verification,
 the client needs to fetch the ROLE.rotate.ID.PREV file
```
this line now feels repetitive

joshuagl commented on tap8.md:
```diff
@@ -268,37 +254,31 @@ to a rotation to a null key is invalid.  The null key is a hard coded value used
 multiple parties, will now be much easier to do.  This will
-provide a simple mechanism extending and shrinking projects by
-themselves without an individual with elevated privileges, but based
+provide a simple mechanism extending and shrinking project membership without an individual with elevated privileges, but based
```
```suggestion
provide a simple mechanism for extending and shrinking project membership without an individual with elevated privileges, but based
```

joshuagl reviewed (commented):
This looks great, thank you. I added a question/clarification but otherwise LGTM.

An additional minor comment: there's some strange line lengths in the edits i.e., line 49 is very long and line 191 is very short. The rest of the file wraps at ~80characters. Should we care about this? If we should, perhaps we could add a linter?

mnm678 commented:
> This looks great, thank you. I added a question/clarification but otherwise LGTM.

Thanks for the review!

> 
> An additional minor comment: there's some strange line lengths in the edits i.e., line 49 is very long and line 191 is very short. The rest of the file wraps at ~80characters. Should we care about this? If we should, perhaps we could add a linter?

I added a commit that cleans this up a bit for this TAP. In the long term it'd be nice to have a consistent stance backed by a linter. The ~80 character lines look nice, but lead to strange diffs when the text is edited, so I'd advocate for line breaks at the sentence or paragraph level.

joshuagl commented on tap8.md:
```diff
@@ -28,38 +28,31 @@ root role. The root role will continue to use the root metadata to establish
 after a rotation to null. The client will detect a rotation to a null key
-and treat it as if the delegation was missing a key. The rest of the
-delegator's file would still be valid.
+and treat it as if the metadata was unsigned.
```
ah, of course. I had misremembered how this works. Thanks!

joshuagl commented:
> LGTM, thanks!
> 
> > I added a commit that cleans this up a bit for this TAP. In the long term it'd be nice to have a consistent stance backed by a linter. The ~80 character lines look nice, but lead to strange diffs when the text is edited, so I'd advocate for line breaks at the sentence or paragraph level.
> 
> Thank you for cleaning these up. Last time I looked (to use in SLSA [slsa-framework/slsa#75](https://github.com/slsa-framework/slsa/issues/75)) I couldn't find a linter which supports semantic line breaks. That was long enough ago that it would be worth another look. Let's file a separate issue for this?

Filed https://github.com/theupdateframework/taps/issues/164

mnm678 commented:
@JustinCappos can you re-approve so this can be merged?
